src/zcash/JoinSplit.hpp

```diff
@@ -78,7 +78,11 @@ class JoinSplit {
         uint64_t vpub_old,
         uint64_t vpub_new,
         const uint256& rt,
-        bool computeProof = true
+        bool computeProof = true,
+        // For paymentdisclosure, we need to retrieve the esk.
+        // Reference as non-const parameter with default value leads to compile error.
+        // So use pointer for simplicity.
+        uint256 *out_ephemeralSecretKey = nullptr
```

Comment:
I'd weakly prefer a `boost::optional` over a `nullptr`.

Reply:
Scratch that - `esk` is being used as a remote reference for setting, not using, so a pointer makes sense.

Reply:
Discussed, leave as pointer.